Framework/API/src/AlgorithmFactory.cpp

```diff
@@ -306,6 +306,30 @@ AlgorithmFactoryImpl::getCategories(bool includeHidden) const {
   return validCategories;
 }
 
+/**
+  * Returns a single algorithm descriptor for the given algorithm name
+  *
+  * @param algName The name of the algorithm to get a descriptor for
+  * @param version The version of the algorithm (default = latest)
+  * @return AlgorithmDescriptor object
+  * @throws std::runtime_error if algorithm does not exist
+  */
+AlgorithmDescriptor
+AlgorithmFactoryImpl::getDescriptor(const std::string &algName,
+                                    int version) const {
+  AlgorithmDescriptor desc;
+  desc.name = algName;
+
+  boost::shared_ptr<IAlgorithm> alg = create(algName, version);
+  auto categories = alg->categories();
+  desc.version = alg->version();
+  // Set the category if there are any
+  desc.category = categories.empty() ? "" : categories.back();
```

Comment:
I think this should do the same thing as the `getDescriptors` or it could be confusing.

Further to this actually I wonder why the descriptor doesn't just hold a vector<std::string> rather than having to create a long `\\`-separated string? It feels like the `category` field is a hangover from when algorithms could only be in one category.

Reply:
I'm also starting to wonder if we need a to split out the descriptor a bit. For our case we don't actually care about the categories or aliases. As you mentioned maybe we could have a more limited struct, something like `AlgorithmKey`, that contained just the name & version fields and have a method to retrieve that? `AlgorithmDescriptor` could inherit this, in a similar manner to `std::tuple` and it's layout would still match the current implementation. What do you think?

Reply:
I assumed it did the same as `getDescriptors`, since that replaces the `category` field of the struct with the most recent category in the for loop. Hence I took the last elements we were passed

Either way the above point it moot if we create a new struct. This would also potentially avoid us having to create new Algorithm objects too, since we are just trying to see what is the latest version Mantid knows about